server/apikeys.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"errors"
+	"log/slog"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type ApiKey struct {
+	ID        uint      `gorm:"primarykey" json:"id"`
+	CreatedAt time.Time `json:"createdAt"`
+	Value     string    `gorm:"not null" json:"value"`
+	UserID    uint      `gorm:"not null" json:"-"`
+	Note      string    `gorm:"not null" json:"note"`
+}
+
+type ApiKeyRequest struct {
+	Note string `json:"note,required"`
+}
+
+func (b *BaseRouter) addAPIKeyRoutes() {
+	apiKeyGrp := b.rg.Group("/apikeys").Use(AuthRequired(nil))
+
+	apiKeyGrp.POST("", func(ctx *gin.Context) {
+		var req ApiKeyRequest
+		if err := ctx.ShouldBindJSON(&req); err != nil {
+			ctx.AbortWithStatusJSON(http.StatusBadRequest, gin.H{
```

Comment:
We have an `ErrorResponse` struct we could use here instead (and in the other handlers), which would keep this a little more standardized with the rest of the handlers:

Example (from routes.go):
```
c.AbortWithStatusJSON(http.StatusBadRequest, ErrorResponse{Error: err.Error()})
```